### PR TITLE
fix: hide actions and notifications from deleted workspaces

### LINKED ISSUE #12561: remove the duplication on the workspace page when a workspace is deleted
remove this duplication on the workspace page when a workspace is deleted. 
image

[Slack Message](https://codercom.slack.com/archives/CJURPL8DN/p1710240121888339?thread_ts=1710240121.888339&cid=CJURPL8DN)

### PR DESCRIPTION
**Pull request by aslilac**
Closes #12561

Don't bother showing the action buttons et al if the workspace has been deleted

### CODE REVIEW

File: site/src/pages/WorkspacePage/WorkspaceTopbar.tsx [hunk 1]
```diff
@@ -106,6 +106,10 @@ export const WorkspaceTopbar: FC<WorkspaceProps> = ({
     allowAdvancedScheduling,
   );
 
+  const isImmutable =
+    workspace.latest_build.status === "deleted" ||
+    workspace.latest_build.status === "deleting";
```
Comment (matifali) [inline site/src/pages/WorkspacePage/WorkspaceTopbar.tsx:111]: will it also hide the retry action in case of a failed delete?

Comment (aslilac) [reply site/src/pages/WorkspacePage/WorkspaceTopbar.tsx:111]: no, the status will switch to `"failed"`

**Review by buenos-nachos — APPROVED** [review]
Looks good! Left a comment, but I think the code is safe to merge as-is

File: site/src/pages/WorkspacePage/WorkspaceTopbar.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aslilac) [reply site/src/pages/WorkspacePage/WorkspaceTopbar.tsx:111]: yeah, let's go with that. :) (it probably is helpful to make sure we don't regress in a big way)

**Pull request merged**